core/shared/src/main/scala/zio/stm/TArray.scala

```diff
@@ -39,10 +98,124 @@ class TArray[A] private (val array: Array[TRef[A]]) extends AnyVal {
         op(acc, a).flatMap(acc2 => new TArray(array.tail).foldM(acc2)(op))
       }
 
+  /** Atomically evaluate the conjunction of a predicate across the members of the array. */
+  final def forall(predicate: A => Boolean): STM[Nothing, Boolean] = exists(a => !predicate(a)).map(!_)
+
+  /** Atomically evaluate the conjunction of an STM predicate across the members of the array. */
+  final def forallM[E](predicate: A => STM[E, Boolean]): STM[E, Boolean] =
+    countM(predicate).map(_ == array.length) //Universal quantification should not be order dependent so the entire array needs to be observed.
+
   /** Atomically performs side-effect for each item in array */
   final def foreach[E](f: A => STM[E, Unit]): STM[E, Unit] =
     this.foldM(())((_, a) => f(a))
 
+  /** The head of the array. */
+  final def head: STM[Nothing, A] =
```

Comment:
I would nuke this one and only keep `headOption`, but instead rename it to `firstOption` for symmetry with `lastOption`.